CHANGES/2585.removal

```diff
@@ -0,0 +1 @@
+Do not create a new resource on `router.add_get(..., allow_head=True)`
```

Comment:
I guess it's improvement, not removal.

Reply:
It's both optimization and breaking change: in aiohttp 2.x the call `router.add_get('/', handler, name='name')` adds two named resources: `name` for GET and `name-head` for `HEAD`.
But both resources are identical.

After PR merging I want add generic optimization for sequential `add_route()` calls if created resource is the same.
```
router.add_get('/', ...)
router.add_post('/', ...)
```
should create the single resource if *path* and *name* are the same.